pallets/elections/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+    pub(super) fn DefaultForSessionPeriod() -> u32 {
+        DEFAULT_SESSION_PERIOD
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn session_period)]
+    pub(super) type SessionPeriod<T: Config> =
+        StorageValue<_, u32, ValueQuery, DefaultForSessionPeriod>;
+
+    #[pallet::type_value]
+    pub(super) fn DefaultForMillisecsPerBlock() -> u64 {
+        DEFAULT_MILLISECS_PER_BLOCK
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn millisecs_per_block)]
+    pub(super) type MillisecsPerBlock<T: Config> =
+        StorageValue<_, u64, ValueQuery, DefaultForMillisecsPerBlock>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub members: Vec<T::AccountId>,
+        pub millisecs_per_block: u64,
+        pub session_period: u32,
+        pub sessions_per_era: SessionIndex,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                members: Vec::new(),
+                millisecs_per_block: DEFAULT_MILLISECS_PER_BLOCK,
+                session_period: DEFAULT_SESSION_PERIOD,
+                sessions_per_era: DEFAULT_SESSIONS_PER_ERA,
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {
+            <Members<T>>::put(&self.members);
+            <MillisecsPerBlock<T>>::put(&self.millisecs_per_block);
+            <SessionPeriod<T>>::put(&self.session_period);
+            <SessionsPerEra<T>>::put(&self.sessions_per_era);
+        }
+    }
+
+    impl<T: Config> Pallet<T> {}
+
+    #[derive(Debug)]
+    pub enum Error {
+        DataProvider(&'static str),
+    }
+
+    impl<T: Config> ElectionProvider<T::AccountId, BlockNumberFor<T>> for Pallet<T> {
+        type Error = Error;
+        type DataProvider = T::DataProvider;
+
+        // We use authority list so the supports don't influance the result
+        // but we still have to calculate them for the sake of eras payouts
+        fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
+            let voters = Self::DataProvider::voters(None).map_err(Error::DataProvider)?;
+            let members = Pallet::<T>::members();
+            let mut supports: BTreeMap<T::AccountId, Support<T::AccountId>> = members
+                .iter()
+                .map(|id| {
+                    (
+                        id.clone(),
+                        Support {
+                            total: 0,
+                            voters: Vec::new(),
+                        },
+                    )
+                })
+                .collect();
+
+            for (voter, vote, targets) in voters {
+                assert!(
+                    targets.len() == 1,
+                    "There should be only one target in one vote"
+                );
+                let member = &targets[0];
+                if !members.contains(member) {
```

Comment:
I think this is unnecessary -- just use the output of `supports.get_mut(member)` to check it.

Reply:
Also, rule of thumb: using `contains` on a `Vec` should ALWAYS raise a red flag :) This does a linear search over the whole vector.

Reply:
Good idea :)

Reply:
Btw isn't it that for small vector sizes (<100) linear search in vec would be faster than logarithmic for tree structures based on key comparison?

Reply:
Absolutely correct, but not sure where the threshold lies -- especially since this is in wasm. Anyway, we want `members` to grow way beyond `10` :)

Reply:
But then, the implementation of the elections will be different ;)

Reply:
Yes. Anyway, this access to the map is necessary, so I think the current version is OK no matter whether membership check for vectors or for BTrees is faster :)